modules/openapi-generator-gradle-plugin/src/test/kotlin/GenerateTaskFromCacheTest.kt

```diff
@@ -0,0 +1,193 @@
+    fun `configFile - different directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeConfigFileTest()
+        runCacheabilityTestUsingDifferentDirectories(gradleVersion, configFileExtensionContents)
+    }
+
+    // ignoreFileOverride tests
+
+    private val ignoreFileOverrideExtensionContents = """
+        generatorName = "kotlin"
+        inputSpec = file("spec.yaml").absolutePath
+        ignoreFileOverride = file(".openapi-generator-ignore").absolutePath
+        """.trimIndent()
+
+    private fun initializeIgnoreFileTest() {
+        projectDir1.resolve(".openapi-generator-ignore").createNewFile()
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `ignoreFileOverride - same directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeIgnoreFileTest()
+        runCacheabilityTestUsingSameDirectory(gradleVersion, ignoreFileOverrideExtensionContents)
+    }
+
+    @Test(dataProvider = "gradle_version_provider")
+    fun `ignoreFileOverride - different directory - openApiGenerate task output should come from cache`(gradleVersion: String) {
+        initializeIgnoreFileTest()
+        runCacheabilityTestUsingDifferentDirectories(gradleVersion, ignoreFileOverrideExtensionContents)
+    }
+
+    // Helper methods & test fixtures
+
+    private fun runCacheabilityTestUsingSameDirectory(gradleVersion: String, extensionContents: String) {
+        // Arrange
+        withProject(extensionContents)
+
+        // Act
+        val result1 = build {
+            withProjectDir(projectDir1)
+            withArguments("--build-cache", "openApiGenerate")
+            withGradleVersion(gradleVersion)
+        }
+
+        val result2 = build {
+            withProjectDir(projectDir1)
+            withArguments("--build-cache", "clean", "openApiGenerate")
+            withGradleVersion(gradleVersion)
+        }
+
+        // Assert
+        assertEquals(TaskOutcome.SUCCESS, result1.task(":openApiGenerate")?.outcome)
+        assertEquals(TaskOutcome.FROM_CACHE, result2.task(":openApiGenerate")?.outcome)
+    }
+
+    private fun runCacheabilityTestUsingDifferentDirectories(gradleVersion: String, extensionContents: String) {
+        // Arrange
+        withProject(extensionContents)
+        projectDir1.copyRecursively(projectDir2)
+
+        // Act
+        val result1 = build {
+            withProjectDir(projectDir1)
+            withArguments("--build-cache", "openApiGenerate")
+            withGradleVersion(gradleVersion)
+        }
+
+        val result2 = build {
+            withProjectDir(projectDir2)
+            withArguments("--build-cache", "openApiGenerate")
+            withGradleVersion(gradleVersion)
+        }
+
+        // Assert
+        assertEquals(TaskOutcome.SUCCESS, result1.task(":openApiGenerate")?.outcome)
+        assertEquals(TaskOutcome.FROM_CACHE, result2.task(":openApiGenerate")?.outcome)
```

Comment:
Something else you might consider is checking all of the outputs exist in the relocated directory as well.  This ensures that not only are you getting a cache hit, but everything that is expected to be cached is in fact being restored (i.e. that all of the outputs are correctly declared).

Reply:
Addressed by 93283eb473b4c008121f9b7acfeb4bba783548c3